### PR TITLE
native: add notebook channel root view

### PR DESCRIPTION
**Pull request by patosullivan**
Adds the basic notebook channel rootview (without the expanded version of the posts for now, would require building out the content rendering component(s) for notebooks unless we just reuse the chat content rendering, which would look pretty weird imo).

Also uses the long press for message actions.

At the moment, the NotebookPost component has a ton of optional boolean props. It probably needs to be split out into separate components, but I'm not sure what those components will be yet. Headline vs summary vs full?

Looks like this:

https://github.com/tloncorp/tlon-apps/assets/1221094/41780c84-19cc-4356-a3bd-685500d4e0eb

### CODE REVIEW

**Review by dnbrwstr — COMMENTED** [review]
This is looking good overall. Few things:
- Could you try to break out the shared pieces between ChatScroll  and NotebookScroll? Seems like there's some overlap there. 
- Could you move the dividers to `NotebookScroll` and use the new `ChannelDivider` component?

File: packages/ui/src/components/Channel/index.tsx [hunk 1]
```diff
@@ -119,7 +117,7 @@ export function Channel({
                       >
                         <Spinner />
                       </View>
-                    ) : (
+                    ) : channel.type === 'chat' ? (
```
Comment (latter-bolden) [inline packages/ui/src/components/Channel/index.tsx:120]: We have to include `dm` and `groupDm` channel types as well. Same for showing the `MessageInput`

File: packages/ui/src/components/Channel/Scroller.tsx [hunk 2]
```diff
@@ -42,6 +44,16 @@ export default function ChatScroll({
   onPressReplies,
   showReplies = true,
 }: {
+  inverted: boolean;
+  postComponent: (props: {
```
Comment (dnbrwstr) [inline packages/ui/src/components/Channel/Scroller.tsx:48]: How bout calling this `renderItem` to mirror the `FlatList` api?

File: packages/ui/src/components/Channel/Scroller.tsx [hunk 3]
```diff
@@ -131,14 +144,15 @@ export default function ChatScroll({
...
+          ) : isFirstPostOfDay && item.type === 'chat' ? (
             <ChannelDivider unreadCount={0} timestamp={item.receivedAt} />
           ) : null}
           <PressableMessage
             ref={activeMessageRefs.current[item.id]}
             isActive={activeMessage?.id === item.id}
           >
-            <ChatMessage
+            {/* ts may yell at you in the editor, but this is fine */}
+            <PostComponent
```
Comment (dnbrwstr) [inline packages/ui/src/components/Channel/Scroller.tsx:155]: And just doing `{renderItem(props)}` here

File: packages/ui/src/components/Channel/index.tsx [hunk 4]
```diff
@@ -86,6 +85,13 @@ export function Channel({
   const title = utils.getChannelTitle(channel);
   const groups = useMemo(() => (group ? [group] : null), [group]);
 
+  const chatChannel =
```
Comment (dnbrwstr) [inline packages/ui/src/components/Channel/index.tsx:88]: Seems like a helper we'll want to reuse -- maybe it could go in logic?

**Review by dnbrwstr — APPROVED** [review]
Couple small nits, but I think it's good to go.

**Review by latter-bolden — APPROVED** [review]

**Pull request merged**